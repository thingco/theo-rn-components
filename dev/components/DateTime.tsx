import React from "react";
import { Text } from "./Base";

import { unixToDateString, unixToTimeString } from "../lib/date-functions";
import { TextStyle, StyleProp } from "react-native";

type DateTimeProps = {
  timestamp: string | number;
  style?: StyleProp<TextStyle>;
};

const FormattedDate = ({ timestamp, ...props }: DateTimeProps) => (
  <Text style={props.style}>{unixToDateString(timestamp)}</Text>
);

const FormattedTime = ({ timestamp, ...props }: DateTimeProps) => (
  <Text style={props.style}>{unixToTimeString(timestamp)}</Text>
);

export { FormattedDate, FormattedTime };
