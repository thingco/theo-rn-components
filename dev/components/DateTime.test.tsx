import React from "react";
import renderer from "react-test-renderer";
import { FormattedDate, FormattedTime } from "./DateTime";

const aug_13_2019__15_29 = 1565706270562;

test("Date renders correctly", () => {
  const tree = renderer
    .create(<FormattedDate timestamp={aug_13_2019__15_29} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Time renders correctly", () => {
  const tree = renderer
    .create(<FormattedTime timestamp={aug_13_2019__15_29} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
