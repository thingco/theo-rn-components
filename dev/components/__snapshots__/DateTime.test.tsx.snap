// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Bariol",
      },
      Object {
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  08/13/2019
</Text>
`;

exports[`Time renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Bariol",
      },
      Object {
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  15:24
</Text>
`;
